performance-tests/bench/report_parser/main.cpp

```diff
@@ -0,0 +1,51 @@
+#include "ArgumentParser.h"
+#include "TimeSeriesGnuPlotFormatter.h"
+#include "TimeSeriesRawFormatter.h"
+
+int parse_time_series(OutputFormat output_format, Report& report,
+  std::ofstream& output_file_stream) {
+  switch (output_format) {
+  case OutputFormat::Gnuplot:
+    TimeSeriesGnuPlotFormatter timeSeriesGnuPlotFormatter;
+    return timeSeriesGnuPlotFormatter.format(report, output_file_stream);
+    break;
+  default:
+    break;
+  }
+
+  TimeSeriesRawFormatter timeSeriesRawFormatter;
+  return timeSeriesRawFormatter.format(report, output_file_stream);
+}
+
+int parse_report(OutputType output_type, OutputFormat output_format,
+  Report& report, std::ofstream& output_file_stream) {
+  switch (output_type) {
+  case OutputType::TimeSeries:
+    return parse_time_series(output_format, report, output_file_stream);
+    break;
+  default:
+    break;
+  }
+
+  return EXIT_FAILURE;
+}
+
+int ACE_TMAIN(int argc, ACE_TCHAR* argv[]) {
+  OutputType output_type = OutputType::None;
+  OutputFormat output_format = OutputFormat::None;
+  Report report{};
+  std::ofstream output_file_stream;
+
+  ArgumentParser argumentParser;
+
+  if (!argumentParser.parse(argc, argv, output_type, output_format,
+    report, output_file_stream)) {
+    return EXIT_FAILURE;
+  }
+
+  if (!parse_report(output_type, output_format, report, output_file_stream)) {
+    return EXIT_SUCCESS;
+  }
+
+  return EXIT_SUCCESS;
```

Comment:
This will always exist successfully now mater what `parse_report` returns.

Do you mean?

```suggestion
  if (parse_report(output_type, output_format, report, output_file_stream)) {
    return EXIT_FAILURE;
  }

  return EXIT_SUCCESS;
```

Or maybe more simply:

```suggestion
  return parse_report(output_type, output_format, report, output_file_stream);
```